src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingFactory.java

```diff
@@ -101,17 +109,33 @@ void validateCreateVirtualTableRequest(CreateTableRequest createVirtualTableRequ
         virtualTableCreationValidator.validateCompatiblePrimaryKeys(virtualTable, physicalTable);
 
         // validate secondary indexes
-        virtualTableCreationValidator.validateAndGetSecondaryIndexMap(virtualTable, physicalTable);
+        Map<DynamoSecondaryIndex, DynamoSecondaryIndex> secondaryIndexMap =
+            virtualTableCreationValidator.validateAndGetSecondaryIndexMap(virtualTable, physicalTable);
+
+        // create the physical table for a new virtual table, or describe the physical table for an existing virtual
+        // table. set the returned physical table description back onto the table mapping, so it includes things that
+        // can only be determined after the physical table is created, like the streamArn.
+        physicalTable = createPhysicalTableIfNotExists
+            ? physicalTableManager.createTableIfNotExists(physicalCreateTableRequest)
```

Comment:
Are we creating physical tables synchronously here?

Reply:
Yes it's still synchronous